README.md

```diff
@@ -1,2 +1,133 @@
+
+# Boostrap extension for the C# Zeebe client
+
+This project is an extension of the [C# Zeebe client project](https://github.com/camunda-community-hub/zeebe-client-csharp). Zeebe Job handlers are automaticly recognized and boostrapped via a [.Net HostedService](https://docs.microsoft.com/en-us/dotnet/architecture/microservices/multi-container-microservice-net-applications/background-tasks-with-ihostedservice).
+
+## Requirements
+
+* .Net 5.0
+* latest [C# Zeebe client release](https://www.nuget.org/packages/zb-client/)
+* latest [Zeebe release](https://github.com/zeebe-io/zeebe/releases/)
+
+## How to use
+
+The Zeebe C# client boostrap extensions is available via nuget (https://www.nuget.org/packages/zb-client-bootsrap/).
+
+Ze [examples](./examples) for more information.
+
+## Quick start
+
+All classes which implement `IJobHandler<T>` or `IAsyncJobHandler<T>` are automaticly found and boostrapped when you register this boostrap project with the `IServiceCollection` extension method `BoostrapZeebe`.
+
+The `BoostrapZeebe` method has two parameters:
+
+1. `ZeebeBootstrapOptions` via [Configuration, Action delegate or both](https://docs.microsoft.com/en-us/dotnet/core/extensions/options-library-authors).
+1. An array with assembly filters, only assemblies which start with one of the filters will be scanned for job handlers.
+
+### Appsettings configuration
+
+```csharp
+ConfigureServices((hostContext, services) => {
+    services.BootstrapZeebe(
+        hostContext.Configuration.GetSection("ZeebeBootstrap"),
+        "SimpleExample"
+    );
+})
+```
+
+**appsettings.json**
+
+```json
+{
+    "ZeebeBootstrap": {
+        "Client": {                
+            "GatewayAddress": "0.0.0.0:26500"
+        },
+        "Worker": {
+            "MaxJobsActive": 1,
+            "TimeoutInMilliseconds": 500,
+            "PollIntervalInMilliseconds": 10000,
+            "PollingTimeoutInMilliseconds": 30000
+        }
+    }
+}
+```
+
+### Delegate configuration
+
+```csharp
+ConfigureServices((hostContext, services) => {
+    services.BootstrapZeebe(
+        options => { 
+            options => { 
+                options.Client = new ClientOptions() {
+                    GatewayAddress = "0.0.0.0:26500"
+                };
+                options.Worker = new WorkerOptions() 
+                {
+                    MaxJobsActive = 1,
+                    TimeoutInMilliseconds = 10000,
+                    PollIntervalInMilliseconds = 30000,
+                    PollingTimeoutInMilliseconds = 1000
+                };
+            }
+        },
+        "SimpleExample"
+    );
+})
+```
+
+### Job
+
+The job is an implementation of `AbstractJob`. A job can be configured via optional attributes. Job types must be unique.
+
+```csharp
+[JobType("SimpleJobV2")]
+[WorkerName("SimpleWorker")]
+[MaxJobsActive(2)]
+[Timeout(500)]
+[PollInterval(10000)]
+[PollingTimeout(500)]
```

Comment:
I'm wondering whether it would make more sense to configure these attributes on the handler instead :)

Reply:
Good comment. I did it because the settings are more Infrastructural then business, so I want them to be as invisible as possible. That being said I also understand your point. I will also ask others about there opinion,

Reply:
@Zelldon can you ask someone to review this project who is also working on zb-client?

Reply:
Sorry but I'm the only maintainer :D 

> Good comment. I did it because the settings are more Infrastructural then business, so I want them to be as invisible as possible. That being said I also understand your point. I will also ask others about there opinion,

This is exactly my thinking which is why I would expect it on the handler. I mean normally you define it on the worker config

Reply:
For now I am going to leave it lie this.